music_assistant/server/providers/deezer/helpers.py

```diff
@@ -0,0 +1,576 @@
+
+
+async def remove_user_albums(client: deezer.Client, album_id: int) -> bool:
+    """Async wrapper of the deezer-python remove_user_albums function."""
+
+    def _get_track():
+        success = client.remove_user_album(album_id=album_id)
+        return success
+
+    return await asyncio.to_thread(_get_track)
+
+
+async def add_user_tracks(client: deezer.Client, track_id: int) -> bool:
+    """Async wrapper of the deezer-python add_user_tracks function."""
+
+    def _get_track():
+        success = client.add_user_track(track_id=track_id)
+        return success
+
+    return await asyncio.to_thread(_get_track)
+
+
+async def remove_user_tracks(client: deezer.Client, track_id: int) -> bool:
+    """Async wrapper of the deezer-python remove_user_tracks function."""
+
+    def _get_track():
+        success = client.remove_user_track(track_id=track_id)
+        return success
+
+    return await asyncio.to_thread(_get_track)
+
+
+async def add_user_artists(client: deezer.Client, artist_id: int) -> bool:
+    """Async wrapper of the deezer-python add_user_artists function."""
+
+    def _get_artist():
+        success = client.add_user_artist(artist_id=artist_id)
+        return success
+
+    return await asyncio.to_thread(_get_artist)
+
+
+async def remove_user_artists(client: deezer.Client, artist_id: int) -> bool:
+    """Async wrapper of the deezer-python remove_user_artists function."""
+
+    def _get_artist():
+        success = client.remove_user_artist(artist_id=artist_id)
+        return success
+
+    return await asyncio.to_thread(_get_artist)
+
+
+async def search_album(client: deezer.Client, query: str, limit: int = 5) -> list[deezer.Album]:
+    """Async wrapper of the deezer-python search_albums function."""
+
+    def _search():
+        result = client.search_albums(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def search_track(client: deezer.Client, query: str, limit: int = 5) -> list[deezer.Track]:
+    """Async wrapper of the deezer-python search function."""
+
+    def _search():
+        result = client.search(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def search_artist(client: deezer.Client, query: str, limit: int = 5) -> list[deezer.Artist]:
+    """Async wrapper of the deezer-python search_artist function."""
+
+    def _search():
+        result = client.search_artists(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def search_playlist(
+    client: deezer.Client, query: str, limit: int = 5
+) -> list[deezer.Playlist]:
+    """Async wrapper of the deezer-python search_playlist function."""
+
+    def _search():
+        result = client.search_playlists(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def get_access_token(mass, app_id, app_secret, code) -> Credential:
```

Comment:
shouldn't the method's name be "update_access_token" if that is what it does ?

Reply:
Fixed in newest commit